cellarium/ml/layers/embedding.py

```diff
@@ -0,0 +1,103 @@
+# Copyright Contributors to the Cellarium project.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from typing import Any
+
+import torch
+from torch import nn
+
+from cellarium.ml.utilities.layers import create_initializer
+
+
+class GeneEmbedding(nn.Module):
```

Comment:
In principle, `GeneExpressionEmbedding` and `MetadataEmbedding` can be unified by allowing either `categorical_vocab_sizes` and `continuous_vocab_size` to be `dict[str, int] | None` (or simply observing that these dictionaries could be empty). Then you could have a `reduction_op` initializer arg, which would be `sum` or `stack`.

Reply:
You could keep the specific classes, but the implementations would be one-liners.

Reply:
One issue in combining them is that the inputs to the forward method have different shapes - `_nd` and `_ncd` which makes it confusing to annotate their shapes with our convention.